src/main.cc

```diff
@@ -392,6 +393,9 @@ usage(void)
             "       -N        Master process runs in foreground and is a worker. No kids.\n"
             "       --foreground\n"
             "                 Master process runs in foreground and creates worker kids.\n"
+            "       --kid role-ID\n"
+            "                 Play a given SMP kid process role, with a given ID. Do not use\n"
+            "                 this option. It is meant for the master process use only.\n"
```

Comment:
If this is truly for internal use only as the text says then this should not be documented at all in the text. Instead use a code comment to prevent anyone re-using it, or thinking its missing text was a mistake.

Reply:
I disagree that internally used command line options should not be documented in usage(): Folks seeing this option in `ps` and similar output will naturally want to know what it is, and `squid -h` is the first place many will look.  This option is a part of the interface, even though it is essentially a read-only part.

There is no harm documenting _all_ command line options here, and there is benefit from doing so.